presto-main/src/test/java/io/prestosql/operator/BenchmarkUnnestOperator.java

```diff
@@ -0,0 +1,424 @@
+                    builder.appendNull();
+                }
+                else {
+                    INTEGER.writeLong(builder, random.nextInt(INT_UPPER_LIMIT));
+                }
+            }
+
+            return builder.build();
+        }
+
+        private Block produceArrayBlock(ArrayType arrType, int entries, double primitiveNulls, double rowElementNulls)
+        {
+            BlockBuilder builder = arrType.createBlockBuilder(null, 100);
+            Type elementType = arrType.getElementType();
+
+            for (int i = 0; i < entries; i++) {
+                int arrLength = minNestedCardinality + random.nextInt(maxNestedCardinality - minNestedCardinality);
+                builder.appendStructure(produceBlock(elementType, arrLength, primitiveNulls, rowElementNulls));
+            }
+            return builder.build();
+        }
+
+        private Block produceMapBlock(MapType mapType, int entries, double primitiveNulls, double rowElementNulls)
+        {
+            int[] mapLengths = new int[entries];
+            int[] offsets = new int[entries + 1];
+
+            int elementCount = 0;
+            for (int i = 0; i < entries; i++) {
+                mapLengths[i] = minNestedCardinality + random.nextInt(maxNestedCardinality - minNestedCardinality);
+                offsets[i + 1] = elementCount;
+                elementCount += mapLengths[i];
+            }
+
+            // No null map objects OR null keys
+            // This stops propagation of null percentages, but for benchmarks this is okay
+            Block keyBlock = produceBlock(mapType.getKeyType(), elementCount, 0.0, 0.0);
+            Block valueBlock = produceBlock(mapType.getValueType(), elementCount, primitiveNulls, rowElementNulls);
+
+            return mapType.createBlockFromKeyValue(Optional.empty(), offsets, keyBlock, valueBlock);
+        }
+
+        public Block produceRowBlock(RowType rowType, int entries, double primitiveNulls, double rowElementNulls)
+        {
+            boolean[] rowIsNull = new boolean[entries];
+            int nonNullCount = 0;
+            for (int i = 0; i < entries; i++) {
+                if (random.nextDouble() < rowElementNulls) {
+                    rowIsNull[i] = true;
+                }
+                else {
+                    rowIsNull[i] = false;
+                    nonNullCount++;
+                }
+            }
+
+            int fieldCount = rowType.getTypeParameters().size();
+            Block[] fieldBlocks = new Block[fieldCount];
+
+            for (int i = 0; i < fieldCount; i++) {
+                fieldBlocks[i] = produceBlock(rowType.getTypeParameters().get(i), nonNullCount, primitiveNulls, rowElementNulls);
+            }
+
+            return RowBlock.fromFieldBlocks(entries, Optional.of(rowIsNull), fieldBlocks);
+        }
+
+        public static String generateRandomString(int length)
+        {
+            byte[] array = new byte[length];
+            new Random().nextBytes(array);
```

Comment:
Why not use the `Random` instance in this class?

Reply:
The random bytes are not necessarily going to be a valid UTF-8 string. It'd be more robust to either 1) generate a string out of simple letters and numbers or 2) generate random codepoints and assemble the string from that, instead.